kategory/src/main/kotlin/kategory/typeclasses/Foldable.kt

```diff
@@ -63,18 +64,16 @@ interface Foldable<F> : Typeclass {
      * Certain structures are able to implement this in such a way that folds can be short-circuited (not traverse the
      * entirety of the structure), depending on the G result produced at a given step.
      */
-    fun <G, A, B> foldM(MG: Monad<G>, fa: HK<F, A>, z: B, f: (B, A) -> HK<G, B>): HK<G, B> {
-        return foldL(fa, MG.pure(z), { gb, a -> MG.flatMap(gb) { f(it, a) } })
-    }
+    fun <G, A, B> foldM(MG: Monad<G>, fa: HK<F, A>, z: B, f: (B, A) -> HK<G, B>): HK<G, B> =
```

Comment:
Sadly it's not possible

<img width="897" alt="screen shot 2017-07-17 at 12 26 42 am" src="https://user-images.githubusercontent.com/514871/28252216-a6b4ec92-6a86-11e7-8ef3-d55ee724c24e.png">